framework/encode/vulkan_state_writer.cpp

```diff
@@ -1552,7 +1575,8 @@ void VulkanStateWriter::WriteImageMemoryState(const VulkanStateTable& state_tabl
         // Perform memory binding.
         const DeviceMemoryWrapper* memory_wrapper = state_table.GetDeviceMemoryWrapper(wrapper->bind_memory_id);
 
-        if (memory_wrapper != nullptr)
+        if ((wrapper->is_swapchain == true && memory_wrapper == nullptr && wrapper->bind_device != nullptr) ||
+            (wrapper->is_swapchain == false && memory_wrapper != nullptr))
```

Comment:
```
if ((wrapper->is_swapchain && memory_wrapper == nullptr && wrapper->bind_device != nullptr) ||
            (!wrapper->is_swapchain && memory_wrapper != nullptr))
```

Reply:
Fixed. Also changed `is_swapchain` to `is_swapchain_image`